From 867a9da20d0c0acf1a0d279fe557fca2f09a84bb Mon Sep 17 00:00:00 2001
From: Stelios Tsampas <loathingkernel@gmail.com>
Date: Mon, 27 Jan 2025 22:42:05 +0200
Subject: [PATCH 1/2] proton: unset DISPLAY only if it is set.

Also wait for wine to exit before reverting the display driver to
winex11. Fixes an issue with UMU, which doesn't wait for the process
to return when starting the application with `wine start /unix`
---
 proton | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/proton b/proton
index 3a0311fc..f7ec4f50 100755
--- a/proton
+++ b/proton
@@ -1776,8 +1776,9 @@ if __name__ == "__main__":
         #wait for wineserver to shut down
         g_session.run_proc([g_proton.wineserver_bin, "-w"])
         winedrv = "x11"
-        if os.environ.get("PROTON_ENABLE_WAYLAND", False) and os.environ["PROTON_ENABLE_WAYLAND"] == "1":
-            del os.environ["DISPLAY"]
+        if os.environ.get("PROTON_ENABLE_WAYLAND", None) == "1":
+            if os.environ.get("DISPLAY", None):
+                del os.environ["DISPLAY"]
             winedrv = "wayland"
         g_session.run_proc(
             [g_proton.wine_bin, "reg.exe", "add", "HKCU\\Software\\Wine\\Drivers", "/v", "Graphics", "/d", f"{winedrv}", "/f"]
@@ -1785,6 +1786,7 @@ if __name__ == "__main__":
         g_session.run_proc([g_proton.wineserver_bin, "-w"])
         #then run
         rc = g_session.run()
+        g_session.run_proc([g_proton.wineserver_bin, "-w"])
         g_session.run_proc(
             [g_proton.wine_bin, "reg.exe", "add", "HKCU\\Software\\Wine\\Drivers", "/v", "Graphics", "/d", "x11", "/f"]
         )
-- 
2.48.1

